src/less.rs

```diff
@@ -3,21 +3,32 @@
 use std::ffi::OsStr;
 use std::process::Command;
 
-pub fn retrieve_less_version(less_path: &dyn AsRef<OsStr>) -> Option<usize> {
+#[derive(Debug, PartialEq)]
+pub enum LessVersion {
+    Less(usize),
+    BusyBox,
+}
+
+pub fn retrieve_less_version(less_path: &dyn AsRef<OsStr>) -> Option<LessVersion> {
     let resolved_path = grep_cli::resolve_binary(less_path.as_ref()).ok()?;
 
     let cmd = Command::new(resolved_path).arg("--version").output().ok()?;
-    parse_less_version(&cmd.stdout)
+    if cmd.status.success() {
+        parse_less_version(&cmd.stdout)
+    } else {
+        parse_less_version(&cmd.stderr)
+    }
 }
 
-fn parse_less_version(output: &[u8]) -> Option<usize> {
-    if !output.starts_with(b"less ") {
-        return None;
+fn parse_less_version(output: &[u8]) -> Option<LessVersion> {
+    match std::str::from_utf8(output) {
+        Ok(version) if version.starts_with("less ") => {
+            let end = (&version[5..]).find(|c: char| !c.is_ascii_digit())? + 5;
+            Some(LessVersion::Less(version[5..end].parse::<usize>().ok()?))
+        }
+        Ok(version) if version.contains("BusyBox ") => Some(LessVersion::BusyBox),
+        _ => None,
     }
-
-    let version = std::str::from_utf8(&output[5..]).ok()?;
-    let end = version.find(|c: char| !c.is_ascii_digit())?;
-    version[..end].parse::<usize>().ok()
 }
```

Comment:
Bonus points if you move this out to a helper function so that the diff is minimized. Something like this:

```rust
fn parse_vanilla_less_version(output: &[u8]) -> Option<usize> { 
    let version = std::str::from_utf8(&output[5..]).ok()?;
    let end = version.find(|c: char| !c.is_ascii_digit())?;
    version[..end].parse::<usize>().ok()
}

Reply:
Or how about put the original function back with the same name and the BusyBox specific code into a separate function? Since BusyBox doesn't accept long options it has to be detected from the stderr output anyway, which was the reason for the if / else block in `retrieve_less_version`. I was going for as concise as possible, but I can see that's probably not a great habit of mine.

```Rust
pub fn retrieve_less_version(less_path: &dyn AsRef<OsStr>) -> Option<LessVersion> {
    let resolved_path = grep_cli::resolve_binary(less_path.as_ref()).ok()?;

    let cmd = Command::new(resolved_path).arg("--version").output().ok()?;
    if cmd.status.success() {
        parse_less_version(&cmd.stdout)  // this function goes back to it's original form
    } else {
        parse_less_busybox(&cmd.stderr)  // diverge here to a new function
    }
}

fn parse_less_busybox(output: &[u8]) -> Option<LessVersion> {
```

Reply:
The smaller diff you can make, the better, because that is easier to review, plus reduces risk of surprising regressions. So keeping `parse_less_version()` unchanged sounds good to me. I think you should call the new function `parse_busybox_less_version()` instead of `parse_less_busybox()` though since "parse less busybox" doesn't fully make semantic sense on its own.

Reply:
Sounds like a good plan. I'm definitely seeing a theme in your comments, but it's a sensible one. I'll get this iterated in the next few days when I have time.